addons/mail/models/mail_thread.py

```diff
@@ -1358,7 +1360,8 @@ def _message_extract_payload(self, message, save_original=False):
                 else:
                     attachments.append((filename or 'attachment', part.get_payload(decode=True)))
 
-        body, attachments = self._message_extract_payload_postprocess(message, body, attachments)
+        if body:
+            body, attachments = self._message_extract_payload_postprocess(message, body, attachments)
```

Comment:
Does `_message_extract_payload_postprocess()` do anything if `body` is empty?  (Should be the case for a message with no `text/` part, right?)

Reply:
It parses the empty string, raising etree XMLSyntaxError. Just so that overrides may still take place on any email, I moved the condition inside the method itself.